### ISSUE TITLE
[🐛] Incompatible types: ReadableMap cannot be converted to WritableMap

### ISSUE THREAD

**Issue by goriverna** [react-native-image-picker/react-native-image-picker#1759]
Install library, compile with android studio, receive error from Android Studio on the file Utils.java:

error: incompatible types: ReadableMap cannot be converted to WritableMap
                assets.pushMap(getImageResponseMap(uri, options, context));

![image](https://user-images.githubusercontent.com/10176638/122719713-e418d400-d26e-11eb-9e74-aac1bf01a6d4.png)

### WORKAROUND

If you follow the suggestion made by android studio you can cast the method as "(WritableMAp)" 

`assets.pushMap((WritableMap) getImageResponseMap(uri, options, context));`

and

`assets.pushMap((WritableMap) getVideoResponseMap(uri, context));`

![image](https://user-images.githubusercontent.com/10176638/122720041-4245b700-d26f-11eb-8773-e020422d3938.png)

- Image Picker version: 4.0.3
- React Native version: 0.60.6
- Platform: Android
- Development Operating System: Mas OS Big Sur (Intel)
- Dev tools: Android Studio: 4.2.1, Compile SDK: 29, Gradle: 6.2, Android Gradle Plugin Version 3.5.3

**Comment by ariva85** [comment]
Same issue here on RN 0.60.6

**Comment by redocs** [comment]
Same problem.

RN 0.60.6

**Comment by Carl0395** [comment]
I change the return of **getImageResponseMap** and **getVideoResponseMap** for WritableMap in node_modules/react-native-image-picker/android/src/main/java/com/imagepicker/Utils.java and fixed for now.

**Comment by goriverna** [comment]
Yes @Carl0395 as my workaround, but we are waiting for a fix asap.

**Comment by Jinxhao18** [comment]
+1

**Comment by Johan-dutoit** [comment]
PR's welcome.

**Comment by jackfiallos** [comment]
there's a PR that fix this issue here
https://github.com/react-native-image-picker/react-native-image-picker/pull/1841

could be good if the PR is merged since is affecting the latest version 4.3.0

**Comment by Johan-dutoit** [comment]
Before merging that PR, I'd like to find out what's causing the issue.

This problem doesn't exist with the example app (on my side anyway), so it would be great if it the root cause can be identified and then we can make an informed decision as to if that PR is correct, or someone's setup is wrong.

**Comment by jackfiallos** [comment]
@Johan-dutoit you're right .. I have checked the code and found this is happening because called methods are using the wrong type, PR with casting is not needed, please check if have a chance https://github.com/react-native-image-picker/react-native-image-picker/pull/1879

**Comment by tarangill** [comment]
Still throwing same error even in version 4.7.1. Please merge the PR into the next release.

**Comment by AbhiSNSoft** [comment]
Almost 1 year & haven't fixed this?

**Comment by LucasFGomes** [comment]
Any concrete solution, please?
